packages/@aws-cdk/aws-ecs/test/ec2/test.ec2-task-definition.ts

```diff
@@ -1006,8 +1006,8 @@ export = {
         Family: 'Ec2TaskDef',
         Volumes: [{
           Name: 'scratch',
-          EfsVolumeConfiguration: {
-            FileSystemId: 'local',
+          EFSVolumeConfiguration: {
```

Comment:
This is causing some weird behavior...

I deployed an older cdk app that generates the following template config:

```json
        "Volumes": [
          {
            "Name": "vol1",
            "EfsVolumeConfiguration": {
              "FileSystemId": {
                "Ref": "MyEfsFileSystem570A3D32"
              },
              "RootDirectory": "/root"
            }
          }
        ]
```

The deployment went through successfully. I then switched to the new version that now generates:

```json
        "Volumes": [
          {
            "EFSVolumeConfiguration": {
              "FilesystemId": {
                "Ref": "MyEfsFileSystem570A3D32"
              },
              "RootDirectory": "/root"
            },
            "Name": "vol1"
          }
        ]
```

And the deployment indicates that a replacement is taking place:

<img width="995" alt="Screen Shot 2020-09-16 at 12 49 52 AM" src="https://user-images.githubusercontent.com/1428812/93268945-c174fe80-f7b6-11ea-82cb-d45f4b978679.png">

It says that the `TaskDef` has been deleted, but I don't see an event for a new `TaskDef` creation. However, the result is that the `TaskDef` isn't actually deleted, and seems to be unaffected:

<img width="876" alt="Screen Shot 2020-09-16 at 12 53 06 AM" src="https://user-images.githubusercontent.com/1428812/93269100-0436d680-f7b7-11ea-84d1-f27a94c1fd05.png">

Does anyone have some light to shed on this? Might be just a CFN quirk and this change is actually has no affect?

`cdk diff` shows:

```console
Stack TaskDef
Resources
[~] AWS::ECS::TaskDefinition TaskDef TaskDef54694570 replace
 └─ [~] Volumes (requires replacement)
     └─ @@ -1,11 +1,11 @@
        [ ] [
        [ ]   {
        [-]     "Name": "vol1",
        [-]     "EfsVolumeConfiguration": {
        [-]       "FileSystemId": {
        [+]     "EFSVolumeConfiguration": {
        [+]       "FilesystemId": {
        [ ]         "Ref": "MyEfsFileSystem570A3D32"
        [ ]       },
        [ ]       "RootDirectory": "/root"
        [-]     }
        [+]     },
        [+]     "Name": "vol1"
        [ ]   }
        [ ] ]

```

Reply:
@iliapolo did you see any change in the ECS console to the tasks themselves during deploy? Generally when a task is updated in CFN it does a rolling deploy so service isn't interrupted. I'm wondering if it does something like that in this case or of its just a total no-op.

Reply:
I haven't actually created any tasks...just the def. I'll try.

Reply:
Checked with a running task. What happens is that a rolling update is performed.
We decided to proceed with this change in spite of the resource replacement because it shouldn't cause downtime and the change is essentially a fix to a bug in the spec that was **manually** introduced in this PR: https://github.com/aws/aws-cdk/pull/8467/files

We don't think it would be wise to remain with this corrupt schema.